<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>CellFuse: Multi-modal integration of single-cell and spatial proteomics data • CellFuse</title>
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="deps/headroom-0.11.0/headroom.min.js"></script><script src="deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="deps/search-1.0.0/fuse.min.js"></script><script src="deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="CellFuse: Multi-modal integration of single-cell and spatial proteomics data">
<meta name="description" content="More about what it does (maybe more than one line). Continuation lines should be indented.">
<meta property="og:description" content="More about what it does (maybe more than one line). Continuation lines should be indented.">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="index.html">CellFuse</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="articles/CellFuse_DataPreparation.html">CellFuse Workflow: Data Preparation and Model Execution</a></li>
    <li><a class="dropdown-item" href="articles/Cytof_CITESeq_DataIntegration.html">Multi-modal integration of CyTOF and CITE-Seq data</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="search.json">
</form></li>
      </ul>
</div>


  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><div class="section level1">
<div class="page-header"><h1 id="cellfuse">CellFuse<a class="anchor" aria-label="anchor" href="#cellfuse"></a>
</h1></div>
<p><strong>CellFuse</strong> is an R package for multimodal single-cell and spatial proteomics integration using supervised contrastive deep learning. Single-cell and spatial proteomic technologies capture complementary biological information; however, no single platform measures all modalities within the same cell. Most existing integration methods (e.g., Seurat, Harmony) are optimized for transcriptomic data and assume extensive shared feature overlap, an assumption that often fails for low-dimensional proteomic modalities.</p>
<div class="section level3">
<h3 id="workflow-overview">Workflow Overview<a class="anchor" aria-label="anchor" href="#workflow-overview"></a>
</h3>
<p>CellFuse operates in three sequential stages:</p>
<ol style="list-style-type: decimal">
<li>
<strong>Model Training</strong> – Learn a shared embedding space using labeled reference data.</li>
<li>
<strong>Cell Type Prediction</strong> – Project query cells into embedding space and assign labels via KNN.</li>
<li>
<strong>Data Integration</strong> – Perform normalization to harmonize modalities.</li>
</ol>
<p><img src="reference/figures/Figure1_v2.png" width="800" height="200"></p>
<p>This work has been led by <a href="https://github.com/AbhivKoladiya" class="external-link">Abhishek Koladiya</a> from <a href="https://kldavislab.org/" class="external-link">Kara Davis Lab</a> @Stanford</p>
</div>
<div class="section level3">
<h3 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h3>
<p>CellFuse relies on Python for deep learning components via the reticulate interface. Before using the package, please configure a Python environment with the required dependencies.</p>
<p>We recommend creating a dedicated conda environment:</p>
<pre class="shell"><code>
# Create the environment
conda create -n myenv python=3.10 -y

# Activate the environment
conda activate myenv

# Install required Python packages
conda install pytorch pandas scikit-learn matplotlib seaborn</code></pre>
<p>After creating the environment, configure it within R:</p>
<pre><code><span></span>
<span><span class="co"># Load and install R dependencies</span></span>
<span><span class="va">required_packages</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"reticulate"</span>, <span class="st">"remotes"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">installed</span> <span class="op">&lt;-</span> <span class="va">required_packages</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/installed.packages.html" class="external-link">installed.packages</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/any.html" class="external-link">any</a></span><span class="op">(</span><span class="op">!</span><span class="va">installed</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="va">required_packages</span><span class="op">[</span><span class="op">!</span><span class="va">installed</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="co"># Use the conda environment from R</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://rstudio.github.io/reticulate/" class="external-link">reticulate</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Use the conda environment</span></span>
<span><span class="fu"><a href="https://rstudio.github.io/reticulate/reference/use_python.html" class="external-link">use_condaenv</a></span><span class="op">(</span><span class="st">"myenv"</span>, required <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Verify Python configuration</span></span>
<span><span class="fu"><a href="https://rstudio.github.io/reticulate/reference/py_config.html" class="external-link">py_config</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Install and load CellFuse:</span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"karadavis-lab/CellFuse"</span><span class="op">)</span></span>
<span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">CellFuse</span><span class="op">)</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="getting-started">Getting Started<a class="anchor" aria-label="anchor" href="#getting-started"></a>
</h2>
<p>The best way to get started with CellFuse is to explore the package’s vignettes and articles (available at <a href="https://karadavis-lab.github.io/CellFuse" class="external-link uri">https://karadavis-lab.github.io/CellFuse</a>).</p>
</div>
<div class="section level2">
<h2 id="citation">Citation<a class="anchor" aria-label="anchor" href="#citation"></a>
</h2>
<div class="sourceCode" id="cb3"><pre class="sourceCode bibtex"><code class="sourceCode bibtex"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a><span class="va">@article</span>{<span class="ot">koladiya2025cellfuse</span>,</span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a>  <span class="dt">title</span>={CellFuse Enables Multi-modal Integration of Single-cell and Spatial Proteomics Data},</span>
<span id="cb3-3"><a href="#cb3-3" tabindex="-1"></a>  <span class="dt">author</span>={Koladiya, Abhishek and Good, Zinaida and Varra, Sricharan Reddy and Bendall, Sean C and Davis, Kara L},</span>
<span id="cb3-4"><a href="#cb3-4" tabindex="-1"></a>  <span class="dt">journal</span>={bioRxiv},</span>
<span id="cb3-5"><a href="#cb3-5" tabindex="-1"></a>  <span class="dt">year</span>={2025},</span>
<span id="cb3-6"><a href="#cb3-6" tabindex="-1"></a>  <span class="dt">publisher</span>={Cold Spring Harbor Laboratory},</span>
<span id="cb3-7"><a href="#cb3-7" tabindex="-1"></a>  <span class="dt">doi</span>={10.1101/2025.07.23.665976},</span>
<span id="cb3-8"><a href="#cb3-8" tabindex="-1"></a>  <span class="dt">url</span>={https://www.biorxiv.org/content/10.1101/2025.07.23.665976v1}</span>
<span id="cb3-9"><a href="#cb3-9" tabindex="-1"></a>}</span></code></pre></div>
</div>
</div>
  </main><aside class="col-md-3"><div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="https://opensource.org/licenses/mit-license.php" class="external-link">MIT</a></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing CellFuse</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Abhishek Koladiya <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0002-7348-4048" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a>  </li>
</ul>
</div>



  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Abhishek Koladiya.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.2.0.</p>
</div>

    </footer>
</div>





  </body>
</html>
